src/MainUtils.cpp

```diff
@@ -233,8 +233,11 @@ string getTargetOptions() {
   string targetOptions = "";
   if (mtriple != "")
     targetOptions = "--mtriple=" + mtriple;
+  // Comand cannot tolerate extra spaces. Add only when needed.
+  if (mtriple != "" && mcpu != "")
+    targetOptions += " ";
   if (mcpu != "")
-    targetOptions += " --mcpu=" + mcpu;
+    targetOptions += "--mcpu=" + mcpu;
```

Comment:
If it cannot tolerate extra spaces then you're adding one if triple isn't empty string but then you never add the triple.

Reply:
I see, it's an AND so it's good but, if perhaps you want to enforce that mcpu should only be used in conjunction with triple then you should use an assert instead to detect the case in which mcpu is used but triple isn't set.

Reply:
There is no requirement that both be used.